File: backend/finance-user/src/main/java/com/finance/backend/service/UserLayoutService.java
package com.finance.backend.service;

import com.fasterxml.jackson.databind.JsonNode;
import com.finance.backend.dto.UserLayoutResponse;
import com.finance.backend.mapper.UserLayoutMapper;
import com.finance.backend.model.UserLayout;
import com.finance.backend.repository.UserLayoutRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.Instant;

@Service
@RequiredArgsConstructor
public class UserLayoutService {

    private final UserLayoutRepository repository;
    private final UserLayoutMapper mapper;

    public UserLayoutResponse getOrEmpty(String userSub) {
        UserLayout layout = repository.findById(userSub).orElseGet(() -> UserLayout.emptyFor(userSub));
        return mapper.toResponse(layout);
    }

    public UserLayoutResponse saveOverview(String userSub, JsonNode overview) {
        UserLayout layout = repository.findById(userSub).orElseGet(() -> UserLayout.emptyFor(userSub));
        layout.setOverview(overview);
        layout.setUpdatedAt(Instant.now());
        return mapper.toResponse(repository.save(layout));
    }
}

File: frontend/src/shared/services/userLayoutService.js
import api from './api';

const LAYOUT_PATH = '/user/layout';

export const userLayoutService = {
  get: async () => {
    const response = await api.get(LAYOUT_PATH);
    return response.data.data;
  },

  saveOverview: async (overview) => {
    const response = await api.put(`${LAYOUT_PATH}/overview`, overview);
    return response.data.data;
  },
};
